### PR TITLE
Schema Store

### PR DESCRIPTION
**Pull request by RyanLassigBanks**
Add schema store triat, and postgres, sqlite, and `HashMap` implementations

### CODE REVIEW

**Comment by chenette** [comment]
The PR description's meaning is clear, but corrections now would be helpful when creating the next release notes. My recommendation: "Add schema store trait and PostgreSQL, SQLite, and hashmap implementations"

P.S. If "hashmap" means a specific language's HashMap (like Rust?), then s/hashmap/HashMap/.

**Review by dplumb94 — APPROVED** [review]

File: sdk/src/grid_db/schemas/store/mod.rs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            number_exponent: model.number_exponent,
+            enum_options: model.enum_options.split(',').map(String::from).collect(),
+            struct_properties: children,
+            service_id: model.service_id,
+        }
+    }
+}
+
+pub trait SchemaStore: Send + Sync {
+    fn add_schema(&self, schema: &Schema) -> Result<(), SchemaStoreError>;
```
Comment (dnewh) [inline sdk/src/grid_db/schemas/store/mod.rs:157]: I think these should have rustdocs similar to the other modules

**Review by dnewh — APPROVED** [review]

**Review by dnewh — APPROVED** [review]

**Review by dnewh — APPROVED** [review]

File: sdk/src/grid_db/schemas/store/mod.rs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    pub properties: Vec<PropertyDefinition>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub service_id: Option<String>,
+    pub start_commit_num: i64,
+    pub end_commit_num: i64,
+}
+
+#[cfg(feature = "diesel")]
+impl Into<(NewGridSchema, Vec<NewGridPropertyDefinition>)> for &Schema {
```
Comment (vaporos) [inline sdk/src/grid_db/schemas/store/mod.rs:39]: Move this into diesel::

**Review by dplumb94 — DISMISSED** [review]

**Review by vaporos — DISMISSED** [review]

File: sdk/src/grid_db/schemas/error.rs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            SchemaStoreError::DuplicateError {
+                context,
+                source: None,
+            } => write!(f, "The element already exists: {}", context),
+            SchemaStoreError::NotFoundError(ref s) => write!(f, "Element not found: {}", s),
+        }
+    }
+}
+
+#[cfg(feature = "diesel")]
```
Comment (dnewh) [inline sdk/src/grid_db/schemas/error.rs:115]: I think these should be in diesel

**Review by dnewh — CHANGES_REQUESTED** [review]

File: sdk/src/grid_db/schemas/error.rs (hunk 3 above)
Comment (RyanLassigBanks) [reply sdk/src/grid_db/schemas/error.rs]: Updatd

**Review by dnewh — DISMISSED** [review]

**Review by dplumb94 — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**